core/service/src/lib.rs

```diff
@@ -489,6 +497,18 @@ impl<Components: components::Components> Service<Components> {
 		let _ = self.to_spawn_tx.unbounded_send(Box::new(task));
 	}
 
+	/// Spawns a task in the background that runs the future passed as
+	/// parameter. The given task is considered infallible, i.e. if it errors we
+	/// trigger a service exit.
+	pub fn spawn_infallible_task(&self, task: impl Future<Item = (), Error = ()> + Send + 'static) {
```

Comment:
bikeshed: i wouldn't say infallible is the right term here. that implies that the task itself cannot fail, where we rather want to convey that we can't tolerate the task failing.

alternate terms: "vital" or "imperative"

Reply:
Yeah! When talking to Pierre he called it "critical" task, which seems adequate.

Reply:
What about the term “essential”?  That seems like the best one.